### PR TITLE
Cleanup all unused code and files

### PR DESCRIPTION
**Pull request by maxlambrecht**
**Pull Request check list**

- [ ] Proper tests/regressions included?
- [ ] Documentation updated?

**Affected functionality**
N/A

**Description of change**
Removing code and files that we are not using in the PoC and that won't probably be used afterwards.

**Which issue this PR fixes**

Signed-off-by: Max Lambrecht <max.lambrecht@hpe.com>

### CODE REVIEW

**Review by jufantozzi — APPROVED** [review]
😍

**Comment by mgbcaio** [comment]
Any specific reason we are excluding the telemetry/prometheus folder? 
In the MVP backlog we are tracking those metrics to be part of Galadriel.

File: datastore.patch [hunk 1]
```diff
@@ -0,0 +1,347 @@
+Index: pkg/common/api.go
```
Comment (jufantozzi) [inline datastore.patch:1]: why add this to this PR?

Comment (maxlambrecht) [reply datastore.patch]: committed by mistake

**Review by mchurichi — APPROVED** [review]
🍙

File: pkg/harvester/endpoints/config.go [hunk 2]
```diff
@@ -15,8 +14,6 @@ type Config struct {
 	// LocalAddr is the local address to bind the listener to.
 	LocalAddress net.Addr
 
-	// Plugin catalog
-	Catalog catalog.Catalog
-
+	Log logrus.FieldLogger
```
Comment (mgbcaio) [inline pkg/harvester/endpoints/config.go:17]: We already have the `Logger logrus.FieldLogger`, what is the purpose of adding this field ?

**Review by mchurichi — APPROVED** [review]

**Pull request merged**